custom/nic_compliance/utils.py

```diff
@@ -0,0 +1,119 @@
+import re
+import base64
+from datetime import timedelta
+from django.conf import settings
+
+from django.contrib.auth.hashers import get_hasher
+
+from dimagi.utils.couch.cache.cache_core import get_redis_client
+
+from corehq.util.view_utils import get_request
+from custom.nic_compliance.const import EXPIRE_LOGIN_ATTEMPTS_IN, REDIS_LOGIN_ATTEMPTS_LIST_PREFIX
+
+PASSWORD_HASHER = get_hasher()
+
+
+def extract_password(obfuscated_password):
+    # Passwords set with expected padding length and format would respect this regex
+    reg_exp = r"^sha256\$([a-z0-9A-Z]{6})([a-zA-Z0-9=]*)([a-z0-9A-Z]{6})=$"
```

Comment:
Should move this to a module-level constant and wrap it in `re.compile(...)`. Then below, call `reg_exp.match(whatever)` instead of `re.match(reg_exp, whatever)`.